tb3_autonomy/src/navigation_behaviors.cpp

```diff
@@ -18,50 +16,75 @@ SetLocations::SetLocations(const std::string& name, const BT::NodeConfig& config
 
 BT::NodeStatus SetLocations::tick()
 {
-    std::string yaml_file = "/overlay_ws/src/tb3_worlds/maps/sim_house_locations.yaml";
-    // ros::param::get("location_file", yaml_file);
-    YAML::Node locations = YAML::LoadFile(yaml_file);
-    int num_locs = locations.size();
-    if (num_locs == 0) {
-        std::cout << "[" << this->name() << "] No locations found." << std::endl;
+    std::string location_file;
+    config().blackboard->get("location_file",location_file);
+
+    try{
+        YAML::Node locations = YAML::LoadFile(location_file);
+        int num_locs = locations.size();
+        if (num_locs == 0) {
+            std::cout << "[" << this->name() << "] No locations found." << std::endl;
+            return BT::NodeStatus::FAILURE;
+        }
+        setOutput("num_locs", num_locs);
+        std::cout << "[" << this->name() << "] Found " << num_locs << " locations." << std::endl;
+
+        std::vector<std::string> location_names{};
+        std::map<std::string, Pose> location_poses{};
+        for(YAML::const_iterator it=locations.begin(); it!=locations.end(); ++it) {
+            auto name = it->first.as<std::string>();
+            location_names.push_back(name);
+            Pose pose = it->second.as<Pose>();
+            location_poses.emplace(name, pose);
+        }
+        setOutput("loc_names", location_names);
+        setOutput("loc_poses", location_poses);
+        
+    } catch(YAML::Exception const& e) {
+        std::cerr << "Couldn't load locations file: " << location_file << ". Error: " << e.what() << std::endl;
         return BT::NodeStatus::FAILURE;
     }
-    setOutput("num_locs", num_locs);
-    std::cout << "[" << this->name() << "] Found " << num_locs << " locations." << std::endl;
+
     return BT::NodeStatus::SUCCESS;
 }
 
 BT::PortsList SetLocations::providedPorts()
 {
-    return { BT::OutputPort<int>("num_locs") };
+    return { BT::OutputPort<int>("num_locs")
+            ,BT::OutputPort<std::vector<std::string> >("loc_names")
+            ,BT::OutputPort<std::map<std::string, Pose>>("loc_poses")
+         };
```

Comment:
The commas are in weird places, please fix

```suggestion
    return { BT::OutputPort<int>("num_locs"),
             BT::OutputPort<std::vector<std::string> >("loc_names"),
             BT::OutputPort<std::map<std::string, Pose>>("loc_poses")
         };
```